app/controllers/instructor-groups.js

```diff
@@ -18,59 +17,46 @@ export default Controller.extend({
   schoolId: null,
   titleFilter: null,
   showNewInstructorGroupForm: false,
-  newInstructorGroups: [],
-  instructorGroups: computed('selectedSchool', function(){
-    let defer = RSVP.defer();
+  newInstructorGroup: null,
+  deletedInstructorGroup: null,
+
+  instructorGroups: computed('selectedSchool', 'deletedInstructorGroup', 'newInstructorGroup', async function(){
     let schoolId = this.get('selectedSchool').get('id');
-    if(isEmpty(schoolId)){
-      defer.resolve([]);
-    } else {
-      this.get('store').query('instructor-group', {
-        filters: {
-          school: schoolId
-        }
-      }).then(instructorGroups => {
-        defer.resolve(instructorGroups);
-      });
+    if(isEmpty(schoolId)) {
+      resolve([]);
     }
-
-    return PromiseArray.create({
-      promise: defer.promise
+    return await this.get('store').query('instructor-group', {
+      filters: {
+        school: schoolId
+      }
     });
   }),
 
   changeTitleFilter: task(function * (value) {
-    const clean = escapeRegExp(value);
-    this.set('titleFilter', clean);
+    this.set('titleFilter', value);
     yield timeout(250);
-
-    return clean;
+    return value;
   }).restartable(),
 
   hasMoreThanOneSchool: gt('model.schools.length', 1),
   filteredInstructorGroups: computed(
     'changeTitleFilter.lastSuccessful.value',
```

Comment:
Since this now depends online on `titleFilter` `'changeTitleFilter.lastSuccessful.value'` should be `'titleFilter'`.